### ISSUE TITLE
RANSAC consumes too much RAM

### ISSUE THREAD

**Issue by yjmantilla** [sappelhoff/pyprep#17]
I have been testing pyprep against our datasets (GRUNECO, University of Antioquia, Colombia). For example I have 58 channels @ 1000Hz for about 5 minutes. This will need like 10 gb of RAM awarded from the OS to python. This is quite a lot for personal computers.

I have two possible proposals to solve this:
- Run RANSAC in a channel wise manner. 
- Automatically identify the resample needed for the data to fit memory 

I like more the first proposal and @sappelhoff  agrees with this approach. This was discussed in #9 .

Once #9 detrend_correction is merged into master I will upload this code so it may be reviewed and tested. It considerably changes find_bad_by_ransac() and a tiny change in run_ransac()

**Comment by sappelhoff** [comment]
looking forward to a PR, I hope this would not slow down RANSAC too much :+1:

**Comment by yjmantilla** [comment]
Im kind of academically compromised right now. I could do a pull request with the code as is, the problem is that is has no explanation. Or maybe by the time @adam2392  is available I will be more available too.

**Comment by yjmantilla** [comment]
Regarding #8 and #9 this is a comment to remind me that I forgot to include another do_detrend = False in another call to noisyChannels in the robust reference function. I will fix this in the channel-wise ransac PR.
